### ISSUE TITLE
SIGSEGV when trying to load cloud config from Secret

### ISSUE THREAD

**Issue by jkbschmid** [kubernetes-sigs/azuredisk-csi-driver#2201]
**What happened**:

When trying to upgrade from v1.28.5 to v1.29.3, the csi-azuredisk-node Pods fail with a segmentation fault.

When specifying and trying to load the cloud-config from Secret, the `fileLoaderConfig` is set to `nil`: https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/fe1355a722f493e098006a71ae7ee1a17447d198/pkg/azureutils/azure_disk_utils.go#L183

This means `loadConfig` is always nil and an `emptyLoader` will be used instead of a `k8sSecretLoader`: https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/61218004d5851b716f5b789a980d9f56f26eace6/vendor/sigs.k8s.io/cloud-provider-azure/pkg/azclient/configloader/load.go#L74

Logs:
```
 $ kubectl -n azuredisk-csi logs csi-azuredisk-node-45fgs -c azuredisk.
I0227 23:40:40.074500       1 azuredisk.go:197] 


Streaming logs below:
I0227 23:40:40.074553       1 azuredisk.go:200] driver userAgent: disk.csi.azure.com/v1.29.3 OSS-helm
I0227 23:40:40.074861       1 azure_disk_utils.go:177] reading cloud config from secret azuredisk-csi/cloud-config-controller
I0227 23:40:40.083570       1 azure_disk_utils.go:196] could not read cloud config from secret azuredisk-csi/cloud-config-controller
I0227 23:40:40.083589       1 azure_disk_utils.go:206] use default AZURE_CREDENTIAL_FILE env var: /etc/kubernetes/azure.json
W0227 23:40:40.083607       1 azure_disk_utils.go:211] load azure config from file(/etc/kubernetes/azure.json) failed with open /etc/kubernetes/azure.json: no such file or directory
I0227 23:40:40.083618       1 azure_disk_utils.go:223] no cloud config provided, error: <nil>, driver will run without cloud config
I0227 23:40:40.083624       1 skus.go:121] NewNodeInfo: Starting to populate node and disk sku information.
W0227 23:40:40.083635       1 azuredisk.go:251] Failed to get node info. Error: NewNodeInfo: Failed to get instance type from Azure cloud provider, nodeName: hubbernetes-node-b595ce8.azure-eastus.github.net, error: no credentials provided for Azure cloud provider
I0227 23:40:40.085660       1 mount_linux.go:282] Detected umount with safe 'not mounted' behavior
I0227 23:40:40.085718       1 driver.go:81] Enabling controller service capability: CREATE_DELETE_VOLUME
I0227 23:40:40.085725       1 driver.go:81] Enabling controller service capability: PUBLISH_UNPUBLISH_VOLUME
I0227 23:40:40.085728       1 driver.go:81] Enabling controller service capability: CREATE_DELETE_SNAPSHOT
I0227 23:40:40.085732       1 driver.go:81] Enabling controller service capability: CLONE_VOLUME
I0227 23:40:40.085735       1 driver.go:81] Enabling controller service capability: EXPAND_VOLUME
I0227 23:40:40.085737       1 driver.go:81] Enabling controller service capability: SINGLE_NODE_MULTI_WRITER
I0227 23:40:40.085741       1 driver.go:100] Enabling volume access mode: SINGLE_NODE_WRITER
I0227 23:40:40.085744       1 driver.go:100] Enabling volume access mode: SINGLE_NODE_READER_ONLY
I0227 23:40:40.085747       1 driver.go:100] Enabling volume access mode: SINGLE_NODE_SINGLE_WRITER
I0227 23:40:40.085752       1 driver.go:100] Enabling volume access mode: SINGLE_NODE_MULTI_WRITER
I0227 23:40:40.085758       1 driver.go:91] Enabling node service capability: STAGE_UNSTAGE_VOLUME
I0227 23:40:40.085763       1 driver.go:91] Enabling node service capability: EXPAND_VOLUME
I0227 23:40:40.085769       1 driver.go:91] Enabling node service capability: GET_VOLUME_STATS
I0227 23:40:40.085772       1 driver.go:91] Enabling node service capability: SINGLE_NODE_MULTI_WRITER
I0227 23:40:40.086003       1 server.go:124] Listening for connections on address: &net.UnixAddr{Name:"//csi/csi.sock", Net:"unix"}
I0227 23:40:40.957621       1 utils.go:77] GRPC call: /csi.v1.Identity/GetPluginInfo
I0227 23:40:40.957634       1 utils.go:78] GRPC request: {}
I0227 23:40:40.958499       1 utils.go:84] GRPC response: {"name":"disk.csi.azure.com","vendor_version":"v1.29.3"}
I0227 23:40:42.139098       1 utils.go:77] GRPC call: /csi.v1.Node/NodeGetInfo
I0227 23:40:42.139115       1 utils.go:78] GRPC request: {}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xc8 pc=0x18d2c5b]

goroutine 116 [running]:
sigs.k8s.io/cloud-provider-azure/pkg/provider.(*Cloud).GetZone(0xc0001d2c00, {0xc0003dd050?, 0xc0002d3000?})
	/mnt/vss/_work/1/go/src/sigs.k8s.io/azuredisk-csi-driver/vendor/sigs.k8s.io/cloud-provider-azure/pkg/provider/azure_zones.go:182 +0x2db
sigs.k8s.io/azuredisk-csi-driver/pkg/azuredisk.(*Driver).NodeGetInfo(0xc000002000, {0x21a8160, 0xc0003dcd80}, 0x26df785?)
	/mnt/vss/_work/1/go/src/sigs.k8s.io/azuredisk-csi-driver/pkg/azuredisk/nodeserver.go:336 +0x13b
github.com/container-storage-interface/spec/lib/go/csi._Node_NodeGetInfo_Handler.func1({0x21a8160, 0xc0003dcd80}, {0x1d33ca0?, 0xc0002b62a0})
	/mnt/vss/_work/1/go/src/sigs.k8s.io/azuredisk-csi-driver/vendor/github.com/container-storage-interface/spec/lib/go/csi/csi.pb.go:7160 +0x72
sigs.k8s.io/azuredisk-csi-driver/pkg/csi-common.logGRPC({0x21a8160, 0xc0003dcd80}, {0x1d33ca0?, 0xc0002b62a0?}, 0xc0002b62c0, 0xc000010960)
	/mnt/vss/_work/1/go/src/sigs.k8s.io/azuredisk-csi-driver/pkg/csi-common/utils.go:80 +0x409
github.com/container-storage-interface/spec/lib/go/csi._Node_NodeGetInfo_Handler({0x1e84560?, 0xc000002000}, {0x21a8160, 0xc0003dcd80}, 0xc00064c780, 0x1fd6b20)
	/mnt/vss/_work/1/go/src/sigs.k8s.io/azuredisk-csi-driver/vendor/github.com/container-storage-interface/spec/lib/go/csi/csi.pb.go:7162 +0x135
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0005c8000, {0x21a8160, 0xc0003dccc0}, {0x21b2240, 0xc0007036c0}, 0xc00044e5a0, 0xc000551980, 0x30ee2c8, 0x0)
	/mnt/vss/_work/1/go/src/sigs.k8s.io/azuredisk-csi-driver/vendor/google.golang.org/grpc/server.go:1343 +0xe03
google.golang.org/grpc.(*Server).handleStream(0xc0005c8000, {0x21b2240, 0xc0007036c0}, 0xc00044e5a0)
	/mnt/vss/_work/1/go/src/sigs.k8s.io/azuredisk-csi-driver/vendor/google.golang.org/grpc/server.go:1737 +0xc4c
google.golang.org/grpc.(*Server).serveStreams.func1.1()
	/mnt/vss/_work/1/go/src/sigs.k8s.io/azuredisk-csi-driver/vendor/google.golang.org/grpc/server.go:986 +0x86
created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 153
	/mnt/vss/_work/1/go/src/sigs.k8s.io/azuredisk-csi-driver/vendor/google.golang.org/grpc/server.go:997 +0x145
```
**What you expected to happen**:

Cloud config being loaded from Secret.

**How to reproduce it**:

Specify no file loader config and use ` --cloud-config-secret-name` and `- --cloud-config-secret-namespace`.

**Anything else we need to know?**: Nope

**Environment**:
```
DRIVER INFORMATION:
-------------------
Build Date: "2024-01-18T14:24:29Z"
Compiler: gc
Driver Name: disk.csi.azure.com
Driver Version: v1.29.3
Git Commit: b1078dc71aa8717ecae7be7abf96b189a4ad8b04
Go Version: go1.21.3
Platform: linux/amd64
Topology Key: topology.disk.csi.azure.com/zone
```

**Comment by andyzhangx** [comment]
@jkbschmid can you try with v1.29.2? we already have a PR to fix this issue on v1.29.4, and v1.29.2 has no such regression, thanks.

**Comment by jkbschmid** [comment]
Using 1.29.2 worked, thanks a lot for the quick response!
